src/App.tsx

```diff
@@ -1,27 +1,93 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import 'bulma/css/bulma.css';
+import classNames from 'classnames';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRed } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [isChoosedType, setIsChoosedType] = useState({
+    all: false,
+    fiveFirst: false,
+    onlyRed: false,
+  });
+  const [getError, setGetError] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleGetGoods = useCallback(async (sortType: string) => {
+    let goodsFromServer: React.SetStateAction<Good[]> = [];
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    try {
+      switch (sortType) {
```

Comment:
You may leave it as it is now, but this solution is not scalable
In case if you pass loader function as parameter to this function - you can create as many functions as you want, but now for that you need to add another case in this function to handle it